packages/sanity/src/structure/comments/src/components/upsell/CommentsUpsellDialog.tsx

```diff
@@ -0,0 +1,97 @@
+import styled from 'styled-components'
+import {CloseIcon, LaunchIcon} from '@sanity/icons'
+import {Box, Stack} from '@sanity/ui'
+import {Button, Dialog} from '../../../../../ui-components'
+import {CommentsUpsellData} from '../../types'
+import {DescriptionSerializer} from 'sanity'
+
+/**
+ * Absolute positioned button to close the dialog.
+ */
+const StyledButton = styled(Button)`
+  position: absolute;
+  top: 12px;
+  right: 12px;
+  z-index: 20;
+  background: transparent;
+  border-radius: 9999px;
+  box-shadow: none;
+  color: white;
+  --card-fg-color: white;
+  :hover {
+    --card-fg-color: white;
+  }
+`
+
+const Image = styled.img`
+  object-fit: cover;
+  width: 100%;
+  height: 100%;
+  height: 200px;
+`
+
+interface CommentsUpsellDialogProps {
+  data: CommentsUpsellData
+  onClose: () => void
+  onPrimaryClick: () => void
+  onSecondaryClick: () => void
+}
+
+export function CommentsUpsellDialog(props: CommentsUpsellDialogProps) {
+  const {data, onClose, onPrimaryClick, onSecondaryClick} = props
+
+  return (
+    <Dialog
+      id="comments-upsell"
+      onClose={onClose}
+      onClickOutside={onClose}
+      __unstable_hideCloseButton
+      bodyHeight="fill"
+      padding={false}
+      footer={{
```

Comment:
Do we want to memoize this value?

Reply:
I don't see anything in the parent components or in this component that will produce unwanted rerenders and would benefit from memoizing this value, given it's not an expensive to calculate. But would love to hear your thoughts on why you consider we shuold.
This is a great post that gave me guidelines to decide to use or not a memo value, https://kentcdodds.com/blog/usememo-and-usecallback#how-is-usememo-different-but-similar .